recipes/lora_dpo_distributed.py

```diff
@@ -143,6 +143,7 @@ def __init__(self, cfg: DictConfig) -> None:
         self.global_step = 0
 
         self._resume_from_checkpoint = cfg.resume_from_checkpoint
+        self._adapter_only = cfg.get("adapter_only", False)
```

Comment:
Can you add a logging INFO statement saying that only the adapter weights will be saved so the user knows right away?

Reply:
Hmmmm, this does make sense to me. I'd much rather have the error propagated from the checkpointer, rather than the recipe file.

I think the complexity arises in how we actually instantiate the checkpointer. I don't think we want `save_adapter_weights_only` as a param on the initialization of the checkpointer b/c the user should be able to take the component and save adapters or the whole file if they want without re-initializing it. Therefore, it has to be a method on `save`.

So if we don't want it being passed to the constructor, we would have to parse out the `save_adapter_weights_only` from the config file before creating the checkpointer, which feels a little messy. Maybe we push the warning to the `save` method of the checkpointer? The only issue with this is that then the user could accidentally have the `save_adapter_weights_only=True`, train their whole model, and then save the checkpoints without actually wanting that feature. But that might be too hand-holdy to worry about?

Thanks for coming on this journey of vomiting all my thoughts on this PR. I think my TL;DR is that this config variable should be separate from the instantiation of the checkpointer, but maybe we push the logging info to the `save` method instead of in the main recipe. 

Thoughts?

Reply:
>  maybe we push the logging info to the save method instead of in the main recipe.

The `save_checkpoint_method` of the checkpointer right? I agree here - we also validate `adapter_only` there.

Reply:
Generally agree with @joecummings here.. the checkpointer class itself should mostly care about stuff that's global across a given fine-tuning run (input and output checkpoint formats). The flag to save only adapter weights is a more local thing since it can in theory vary depending on where we are in training. So imo it makes sense to expose only in `save_checkpoint` and not in init.

A separate (but still relevant) point: our checkpointer naming still seems to imply it's only used for full fine-tuning. This is confusing and another reason why it'd look weird to put `save_adapter_weights_only` in its init. (I don't think that's a good reason to keep `save_adapter_weights_only` outside the checkpointer init, I actually think we should just rename the checkpointer instead)